lib/Sema/CSBindings.cpp

```diff
@@ -1251,6 +1251,7 @@ void PotentialBindings::infer(Constraint *constraint) {
   case ConstraintKind::FunctionInput:
   case ConstraintKind::FunctionResult:
   case ConstraintKind::OpaqueUnderlyingType:
+  case ConstraintKind::PropertyWrapper:
```

Comment:
Just like for member references we'd need to make sure that `wrappedValueType` is not bound until `wrapperType` is resolved. You'd have to adjust inference logic and add `PropertyWrapper` constraint to delaying list for the wrapped value e.g. https://github.com/apple/swift/pull/36568/files#diff-f5d7e33c5b6e1d19ce061b6b45945fb3dd74943f11d2e7d61a6886d26405e9f6R1317

Reply:
Whoops, you're right, thank you! This now behaves the same way as the other member constraints (https://github.com/apple/swift/pull/36568/commits/ee2a8fce16c34df182850074a8b1d5fecb69dab0)

Reply:
Great!